code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -200,6 +218,14 @@
 	use_power(power_usage)
 	update_parents()
 
+/obj/machinery/atmospherics/components/unary/thermomachine/anomaly_act(mob/living/user, obj/item/assembly/signaler/anomaly/pyro)
```

Comment:
What's ``anomaly_act``? This should be on attackby and check for the anomaly type you want

Reply:
Fixed